### PR TITLE
Remove String::Duplicate

### PR DESCRIPTION
**Pull request by ZehMatt**
That's the last of that

### CODE REVIEW

File: src/openrct2/scenario/ScenarioRepository.cpp [hunk 1]
```diff
@@ -623,8 +628,8 @@ class ScenarioRepository final : public IScenarioRepository
             for (uint32_t i = 0; i < numHighscores; i++)
             {
                 ScenarioHighscoreEntry* highscore = InsertHighscore();
-                highscore->fileName = fs.ReadString();
-                highscore->name = fs.ReadString();
+                highscore->fileName = fs.ReadStdString();
```
Comment (duncanspumpkin) [inline src/openrct2/scenario/ScenarioRepository.cpp:631]: Will this cause invalidation of the file format? Or does it just return a std string but internally its the same.

Comment (ZehMatt) [reply src/openrct2/scenario/ScenarioRepository.cpp:631]: Both versions wrote a null terminated sequence, this just results in std::string

File: src/openrct2-ui/windows/ScenarioSelect.cpp [hunk 2]
```diff
@@ -623,16 +623,15 @@ static void WindowScenarioselectScrollpaint(WindowBase* w, DrawPixelInfo* dpi, i
...
 
                     // Draw completion score
-                    const utf8* completedByName = "???";
-                    if (!String::IsNullOrEmpty(scenario->Highscore->name))
+                    u8string completedByName = "???";
+                    if (!scenario->Highscore->name.empty())
                     {
                         completedByName = scenario->Highscore->name;
                     }
-                    SafeStrCpy(buffer, completedByName, 64);
```
Comment (CookiePLMonster) [inline src/openrct2-ui/windows/ScenarioSelect.cpp:631]: Is `buffer` still used elsewhere after this?

Comment (Broxzier) [reply src/openrct2-ui/windows/ScenarioSelect.cpp:631]: `buffer` is used earlier in the code (declared in line 603 and used in line 609). It doesn't serve a purpose after that anymore.

Comment (ZehMatt) [reply src/openrct2-ui/windows/ScenarioSelect.cpp:631]: I will scrape the other raw string uses eventually

**Comment by karst** [comment]
The game crashed for me on achieving a new high score with this build.

**Comment by karst** [comment]
I used the testpark/scenario provided in https://github.com/OpenRCT2/OpenRCT2/issues/19243

**Comment by ZehMatt** [comment]
> I used the testpark/scenario provided in #19243

Thanks, I'll look into it, my own test run was fine oddly.

**Comment by ZehMatt** [comment]
I've applied a fix, this has to be refactored later to pass u8string as the argument and perhaps change the signature also for when no name is supplied, tested this again myself and all seems to be working now.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**